File: Codeforces/479A - Expression.cpp
// ==================================================
// Problem  :   479A - Expression
// Run time :   0.015 sec.
// Language :   C++11
// ==================================================

#include <cstdio>
#include <algorithm>
using namespace std;

int main()
{
    int a, b, c;

    while(scanf("%d %d %d", &a, &b, &c) != EOF) {
        int mx = a+b+c;
        mx = max(mx, a+b*c);
        mx = max(mx, (a+b)*c);
        mx = max(mx, a*b+c);
        mx = max(mx, a*(b+c));
        mx = max(mx, a*b*c);

        printf("%d\n", mx);
    }

    return 0;
}

File: Codeforces/479A - Expression.py
# ==================================================
# Problem  :   479A - Expression
# Run time :   0.061 sec.
# Language :   Python 3.5
# ==================================================

from sys import stdin, stdout

def main():
	a, b, c = tuple(map(int, stdin.read().split()))

	stdout.write(str(max(a+b+c, a+b*c, (a+b)*c, a*b+c, a*(b+c), a*b*c)) + '\n')

if __name__ == '__main__':
	main()
